packages/cli/tests/commonlib/simpleAuthValidator.ts

```diff
@@ -27,83 +32,133 @@ export interface ISimpleAuthObject {
-  ) {
+  public async validate() {
     console.log("Start to validate Simple Auth.");
 
-    const resourceName: string = simpleAuthObject.endpoint.slice(8, -18);
+    const resourceName: string = this.simpleAuthObject.endpoint.slice(8, -18);
     chai.assert.exists(resourceName);
 
     const tokenProvider = MockAzureAccountProvider;
     const tokenCredential = await tokenProvider.getAccountCredentialAsync();
     const token = (await tokenCredential?.getToken())?.accessToken;
 
     console.log("Validating app settings.");
+    const activeResourcePlugins = await getActivePluginsFromProjectSetting(this.projectPath);
+    chai.assert.isArray(activeResourcePlugins);
     const response = await getWebappSettings(
       this.subscriptionId,
       this.rg,
       resourceName,
       token as string
     );
     chai.assert.exists(response);
-    chai.assert.equal(aadObject.clientId, response[PropertiesKeys.clientId]);
-    // chai.assert.equal(aadObject.clientSecret, response[PropertiesKeys.clientSecret]);
-    chai.assert.equal(aadObject.applicationIdUris, response[PropertiesKeys.identifierUri]);
-    chai.assert.equal(aadObject.oauthAuthority, response[PropertiesKeys.oauthAuthority]);
     chai.assert.equal(
-      `${aadObject.oauthAuthority}/v2.0/.well-known/openid-configuration`,
-      response[PropertiesKeys.aadMetadataAddreass]
+      response[PropertiesKeys.clientId],
+      this.ctx[PluginId.Aad][StateConfigKey.clientId]
+    );
+    chai.assert.equal(
+      response[PropertiesKeys.clientSecret],
+      await this.getM365ClientSecret(activeResourcePlugins)
+    );
+    chai.assert.equal(
+      response[PropertiesKeys.identifierUri],
+      this.getExpectedM365ApplicationIdUri(this.ctx, activeResourcePlugins)
+    );
+    chai.assert.equal(
+      response[PropertiesKeys.oauthAuthority],
+      this.ctx[PluginId.Aad][StateConfigKey.oauthAuthority]
+    );
+    chai.assert.equal(
+      response[PropertiesKeys.aadMetadataAddreass],
+      `${
+        this.ctx[PluginId.Aad][StateConfigKey.oauthAuthority]
+      }/v2.0/.well-known/openid-configuration`
     );
 
     console.log("Validating app service plan.");
-    const servicePlanName = isMultiEnvEnabled
-      ? resourceName.replace("-webapp", "-serverfarms")
-      : resourceName;
+    const servicePlanName = resourceName.replace("-webapp", "-serverfarms");
     const serivcePlanResponse = await getWebappServicePlan(
       this.subscriptionId,
       this.rg,
       servicePlanName,
       token as string
     );
-    chai.assert(serivcePlanResponse, servicePlan);
+    const expectedServicePlan =
+      (await getProvisionParameterValueByKey(
+        this.projectPath,
+        this.env,
+        provisionParametersKey.simpleAuthSku
+      )) ?? "F1";
+    chai.assert(serivcePlanResponse, expectedServicePlan);
 
     console.log("Successfully validate Simple Auth.");
   }
+
+  private getExpectedM365ApplicationIdUri(ctx: any, activeResourcePlugins: string[]): string {
```

Comment:
minor this may be extracted to a common function

Reply:
Will consider move the function to common utility file.